spec.html

```diff
@@ -26367,14 +26461,16 @@ <h1>
 
             <emu-alg>
               1. Assert: _state_.[[IsLoading]] is *true*.
-              1. If _module_ is a Cyclic Module Record, _module_.[[Status]] is ~new~, and _state_.[[Visited]] does not contain _module_, then
+              1. If _module_ is a Cyclic Module Record, _module_.[[Status]] is ~new~, _state_.[[Visited]] does not contain _module_, and _loadType_ is ~recursive-load~, then
                 1. Append _module_ to _state_.[[Visited]].
                 1. Let _requestedModulesCount_ be the number of elements in _module_.[[RequestedModules]].
                 1. Set _state_.[[PendingModulesCount]] to _state_.[[PendingModulesCount]] + _requestedModulesCount_.
-                1. For each String _required_ of _module_.[[RequestedModules]], do
-                  1. If _module_.[[LoadedModules]] contains a Record whose [[Specifier]] is _required_, then
+                1. For each ModuleRequest Record _required_ of _module_.[[RequestedModules]], do
+                  1. If _module_.[[LoadedModules]] contains a Record whose [[Specifier]] is _required_.[[Specifier]], then
                     1. Let _record_ be that Record.
-                    1. Perform InnerModuleLoading(_state_, _record_.[[Module]]).
+                    1. Let _innerLoadType_ be ~recursive-load~.
+                    1. If _required_.[[Phase]] is ~source~, set _innerLoadType_ to ~single~.
```

Comment:
```suggestion
                    1. If _required_.[[Phase]] is ~source~, let _innerLoadType_ be ~single~; otherwise let _innerLoadType_ be ~recursive-load~.
```